tests/bluetooth/df/src/common.c

```diff
@@ -53,3 +61,62 @@ void common_delete_adv_set(void)
 	err = bt_le_ext_adv_delete(g_adv);
 	zassert_equal(err, 0, "Failed to delete advertiser set");
 }
+
+void common_set_cl_cte_tx_params(void)
+{
+	uint8_t ant_ids[] = { 0x1, 0x2, 0x3, 0x4, 0x5};
+
+	struct bt_hci_cp_le_set_cl_cte_tx_params *cp;
+	uint8_t *dest_ant_ids;
+	struct net_buf *buf;
+	int err;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_SET_CL_CTE_TX_PARAMS,
+				sizeof(*cp) + ARRAY_SIZE(ant_ids));
+	zassert_not_null(buf, "Failed to create HCI cmd object");
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = g_adv->handle;
+	cp->cte_len = g_cte_len;
+	cp->cte_type = BT_HCI_LE_AOD_CTE_2US;
+	cp->cte_count = 5;
+
+	dest_ant_ids = net_buf_add(buf, ARRAY_SIZE(ant_ids));
+	memcpy(dest_ant_ids, ant_ids, ARRAY_SIZE(ant_ids));
+
+	cp->switch_pattern_len = ARRAY_SIZE(ant_ids);
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_LE_SET_CL_CTE_TX_PARAMS,
+				   buf, NULL);
```

Comment:
Shouldn't something like this be implemented in hci_core.c and called from here instead?

Reply:
There is a function that does the same thing `hci_df_set_conn_cte_tx_param` but it does additional parameters validation that is not necessary here and the function is static (implemented in `direciton.c`). 
There is a public API function `bt_df_set_adv_cte_tx_param` (implemented in the same file) but this one uses atomics etc. to check if valid sequence of function calls is kept. 

If I go with the `bt_df_set_adv_cte_tx_param` then the tests will start to look like sample application not unit test of controller functionality.